bsp/worker/src/mill/bsp/worker/MillBuildServer.scala

```diff
@@ -592,9 +592,8 @@ private class MillBuildServer(
   override def buildTargetCleanCache(cleanCacheParams: CleanCacheParams)
       : CompletableFuture[CleanCacheResult] =
     completable(s"buildTargetCleanCache ${cleanCacheParams}") { state =>
-      val targetIds = state.rootModules.map { case b: BspModule => state.bspIdByModule(b) }
       val (msg, cleaned) =
-        cleanCacheParams.getTargets.asScala.filter(targetIds.contains).foldLeft((
```

Comment:
Instead or removing this filter completely, we could log, that we clean the meta-build which will result in a slow subsequent build request.